src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -666,6 +675,24 @@ protected SAMFileHeader getHeaderForSAMWriter(){
     }
 
     /**
+     * @return If addOutputVCFCommandLine is set, a set of VCF header lines containing the GATK tool name, version,
+     * date and command line, otherwise an empty set.
+     */
+    protected Set<VCFHeaderLine> getDefaultToolVCFHeaderLines() {
+        final Set<VCFHeaderLine> gatkToolHeaderLines = new HashSet<>();
+        if (addOutputVCFCommandLine) {
+            final Map<String, String> simpleHeaderLineMap = new HashMap<>(4);
+            simpleHeaderLineMap.put("ID", this.getClass().getSimpleName());
+            simpleHeaderLineMap.put("Version", getVersion());
+            simpleHeaderLineMap.put("Date", Utils.getDateTimeForDisplay((ZonedDateTime.now())));
+            simpleHeaderLineMap.put("CommandLine", getCommandLine());
+            gatkToolHeaderLines.add(new VCFHeaderLine("source", this.getClass().getSimpleName()));
+            gatkToolHeaderLines.add(new VCFSimpleHeaderLine("GATKCommandLine", simpleHeaderLineMap));
```

Comment:
Could the ID "GATKCommandLine" be retrieve for an overridable method? It will be cool for other toolkits that wants to use the `GATKTool` interface for traversal but want to distinguish their command line into the header...